components/director/internal/domain/eventing/service.go

```diff
@@ -0,0 +1,188 @@
+package eventing
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/label"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/apperrors"
+
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/kyma-incubator/compass/components/director/internal/labelfilter"
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/internal/tenant"
+	"github.com/pkg/errors"
+)
+
+const (
+	RuntimeEventingURLLabel = "runtime/event_service_url"
+	EmptyEventingURL        = ""
+)
+
+//go:generate mockery -name=RuntimeRepository -output=automock -outpkg=automock -case=underscore
+type RuntimeRepository interface {
+	List(ctx context.Context, tenant string, filter []*labelfilter.LabelFilter, pageSize int, cursor string) (*model.RuntimePage, error)
+	GetOldestForFilters(ctx context.Context, tenant string, filter []*labelfilter.LabelFilter) (*model.Runtime, error)
+}
+
+//go:generate mockery -name=LabelRepository -output=automock -outpkg=automock -case=underscore
+type LabelRepository interface {
+	GetByKey(ctx context.Context, tenant string, objectType model.LabelableObject, objectID, key string) (*model.Label, error)
+	Upsert(ctx context.Context, label *model.Label) error
+}
+
+type service struct {
+	runtimeRepo RuntimeRepository
+	labelRepo   LabelRepository
+}
+
+func NewService(runtimeRepo RuntimeRepository, labelRepo LabelRepository) *service {
+	return &service{
+		runtimeRepo: runtimeRepo,
+		labelRepo:   labelRepo,
+	}
+}
+
+func (s *service) GetForApplication(ctx context.Context, appID uuid.UUID) (*model.ApplicationEventingConfiguration, error) {
+	_tenant, err := tenant.LoadFromContext(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while loading tenant from context")
+	}
+
+	var foundDefault, foundOldest bool
+	runtime, foundDefault, err := s.getDefaultRuntimeForAppEventing(ctx, _tenant, appID)
```

Comment:
here we can return runtime that is in another scenario than the app, is it ok?

Reply:
Yes, if the runtime during eventing assignment was in the same scenario as the app and later was removed from that scenario. I'm wondering whether we should do the removal here or handle this in the runtime to scenario assignment?